plugins/inputs/redis_sentinel/redis_sentinel_test.go

```diff
@@ -0,0 +1,391 @@
+package redis_sentinel
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+
+	"github.com/stretchr/testify/require"
+)
+
+const masterName = "mymaster"
+
+func TestRedisSentinelConnect(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	addr := fmt.Sprintf(testutil.GetLocalHost() + ":26379")
+
+	r := &RedisSentinel{
+		Servers: []string{addr},
+	}
+
+	var acc testutil.Accumulator
+
+	err := acc.GatherError(r.Gather)
+	require.NoError(t, err)
+}
+
+func TestRedisSentinelMasters(t *testing.T) {
+	t.Logf("Redis Sentinel: 'sentinel masters <name>'")
+
+	now := time.Now()
+
+	globalTags := map[string]string{
+		"port":   "6379",
+		"source": "redis.io",
+	}
+
+	expectedTags := map[string]string{
+		"port":        "6379",
+		"source":      "redis.io",
+		"master_name": masterName,
+	}
+
+	// has-quorum is a custom field
+	expectedFields := map[string]interface{}{
+		"config-epoch":            0,
+		"down-after-milliseconds": 30000,
```

Comment:
Let's convert tag and field keys to snake case, rather than use hyphens.

Reply:
Is that a hard requirement? Those values come directly from Redis.

Reply:
I believe it's a practice we're trying to [enforce](https://github.com/influxdata/telegraf/wiki/Review#metric-schema) on all new plugins

Reply:
Fixed in 7b31e2a202